src/coreclr/vm/method.hpp

```diff
@@ -2280,6 +2283,13 @@ class MethodDescChunk
         LIMITED_METHOD_CONTRACT;
         m_next = chunk;
     }
+
+    void SetLoaderModuleAttachedToChunk(Module* pModule)
+    {
+        m_flagsAndTokenRange |= enum_flag_LoaderModuleAttachedToChunk;
+        TADDR ppLoaderModule = dac_cast<TADDR>(this) + SizeOf() - sizeof(PTR_Module);
+        *(Module**)ppLoaderModule = pModule;
```

Comment:
Do you think we will need to do this trailing chunk data trick in future situations?  the MethodDesc extra pointers are very tricky to think about and this is heading in the same direction, but at least there it's somewhat abstracted away with the macros that give the MethodDesc sizes for a particular set of flags. (although in the cdacreader i don't bother with a static lookup table and just use popcount on the masked bits to figure out the ordinal for each extra pointer).

My point is maybe it's worth doing something a little bit cleaner here right off the bat.

Reply:
I don't think we'll need anything more added to `MethodDescChunk`. These are famous last words, but I think its pretty good.